src/main/java/cryptator/tree/CryptaEvaluation.java

```diff
@@ -8,81 +8,84 @@
 
-import java.math.BigInteger;
-import java.util.ArrayDeque;
-import java.util.Deque;
-
 import cryptator.solver.CryptaSolutionException;
 import cryptator.specs.ICryptaEvaluation;
 import cryptator.specs.ICryptaNode;
 import cryptator.specs.ICryptaSolution;
 import cryptator.specs.ITraversalNodeConsumer;
 
+import java.math.BigInteger;
+import java.util.ArrayDeque;
+import java.util.Deque;
+
 public class CryptaEvaluation implements ICryptaEvaluation {
 
-	@Override
-	public BigInteger evaluate(ICryptaNode cryptarithm, ICryptaSolution solution, int base) throws CryptaEvaluationException {
-		final EvaluationConsumer evaluationNodeConsumer = new EvaluationConsumer(solution, base);
-		TreeTraversals.postorderTraversal(cryptarithm, evaluationNodeConsumer);
-		return evaluationNodeConsumer.eval();
-	}	
+    @Override
+    public BigInteger evaluate(ICryptaNode cryptarithm, ICryptaSolution solution, int base) throws CryptaEvaluationException {
+        final EvaluationConsumer evaluationNodeConsumer = new EvaluationConsumer(solution, base);
+        TreeTraversals.postorderTraversal(cryptarithm, evaluationNodeConsumer);
+        return evaluationNodeConsumer.eval();
+    }
 
-	private static class EvaluationConsumer implements ITraversalNodeConsumer {
+    private static class EvaluationConsumer implements ITraversalNodeConsumer {
 
-		private final ICryptaSolution solution;
+        private final ICryptaSolution solution;
 
-		private final int base;
+        private final int base;
 
-		private final Deque<BigInteger> stack = new ArrayDeque<>();
+        private final Deque<BigInteger> stack = new ArrayDeque<>();
 
-		private CryptaEvaluationException exception;
+        private CryptaEvaluationException exception;
 
-		public EvaluationConsumer(ICryptaSolution solution, int base) {
-			super();
-			this.solution = solution;
-			this.base = base;
-		}
+        public EvaluationConsumer(ICryptaSolution solution, int base) {
+            super();
+            this.solution = solution;
+            this.base = base;
+        }
 
-		private BigInteger getWordValue(ICryptaNode node) throws CryptaEvaluationException {
-			try {
-				BigInteger v = BigInteger.ZERO;
-				final BigInteger b = BigInteger.valueOf(base);
-				for (char c : node.getWord()) {
-					final int digit = solution.getDigit(c);
-					if(digit < 0 || digit >= base) throw new CryptaEvaluationException("cannot evaluate because of an invalid digit for the evaluation base.");
-					v = v.multiply(b).add(BigInteger.valueOf(digit));
-				}
-				return v;
-			} catch (CryptaSolutionException e) {
-				throw new CryptaEvaluationException("Cannot use a partial solution for evaluation", e);
-			}
-		}
+        private BigInteger getWordValue(ICryptaNode node) throws CryptaEvaluationException {
+            try {
+                BigInteger v = BigInteger.ZERO;
+                final BigInteger b = BigInteger.valueOf(base);
+                if (node instanceof CryptaConstant constant)
```

Comment:
You have chosen to add an attribute to the class. 
Do not use both the attribute and instanceof.

I know that it allows to access the getConstant method, but we must find another way.